chalice/awsclient.py

```diff
@@ -140,8 +198,12 @@ def update_function(self,
         the targeted lambda function.
         """
         lambda_client = self._client('lambda')
-        return_value = lambda_client.update_function_code(
-            FunctionName=function_name, ZipFile=zip_contents)
+        try:
+            return_value = lambda_client.update_function_code(
+                FunctionName=function_name, ZipFile=zip_contents)
+        except Exception as e:
```

Comment:
Offline: Catch LambdaClientError and botocore.exceptions.ClientError here instead.